lib/ui/painting.dart

```diff
@@ -1684,21 +1681,34 @@ class Codec extends NativeFieldWrapperClass2 {
   /// * -1 for infinity repetitions.
   int get repetitionCount native 'Codec_repetitionCount';
 
+  FrameInfo _cachedFrame;
+
   /// Fetches the next animation frame.
   ///
   /// Wraps back to the first frame after returning the last frame.
   ///
   /// The returned future can complete with an error if the decoding has failed.
-  Future<FrameInfo> getNextFrame() {
-    return _futurize(_getNextFrame);
+  Future<FrameInfo> getNextFrame() async {
+    if (_cachedFrame != null && frameCount == 1) {
+      return _cachedFrame;
+    }
+    final Image image = Image._();
+    print('About to call _futureize with image');
+    final int durationMilliseconds = await _futurize((_Callback<int> callback) => _getNextFrame(image, callback));
+    print(image);
+    return _cachedFrame = FrameInfo._(durationMilliseconds, image);
```

Comment:
I think this would be more readable with the assignment separated from the `return`

Reply:
Flipped the meaning of the if statement so we just always return `_cachedFrame` at the end and only assign it if needed.